### DISCUSSION TITLE
xDrip averaging removal

### DISCUSSION THREAD

**Discussion by enpa23** [jakobheyman/xdripJH#1] [General]
Hi, thanks for opening this section.

So, as I wrote on xDrip, being forced to use Juggluco with FSL3 and therefore to set "Libre2 patched app" as data source to get data on xDrip, I noticed that the averaging done by xDrip results in a significant delay when bg values are changing "fast" (  ie ~15 mg/dl over 5 minutes ).  As far as I understand, juggluco is using the original algorithm so the values it provides are the same as  the fsl3.
It could happen that during a bg fall juggluco/fsl3 are giving a 119 mg/dl value, a fingerprick test 113 and xDrip is stuck at 160 because it's averaging the last 20 or so values.
I think that smoothing values could be ok as it avoids to go crazy with sudden random variations as temporary compression lows, but it has to be reduced in order to closer match data provided by the abt algo ( which all in all does its job fairly well ).

The ideal goal for a faster response would be to have the possibility in settings to:
- disable/reduce the averaging ( eg  from latest 20 points to 10 or 5 )
- ( nice to have ) Enable both slope and intercept calibration for that data source, as currently calibration affects the intercept only.

If I'm correct your version of xDrip has removed the averaging, so it would be helpful to me to be pointed to the files that have to be patched in order to get the same result on the original xDrip app so to build it by myself and test its behaviour.

**Comment by jakobheyman** [comment]
Sorry for my slow reply!

In my modified version I have changed the default calibration (if using OOP2 - I don't know how it works if using the patched Libre) and I believe that has removed some delay. However, these changes have been carried out over quite some time and I don't know where in the original code to address the delay that comes with the default xDrip+ calibration.

I am sorry to not have a good reply to your question and I really understand your wish to get rid of the averaging. If you wish, I can try to fix xdripJH so that one can set the duration between each upload of data (I guess this should not be too difficult but I might be wrong) so that you can get the minute data in the phone but only send out every fifth value (or whatever number you wish).

**Reply by enpa23** [reply]
Thanks for your reply. 
No need to sorry, we all have our lives. I was busy in these days too and I had no time to give a look at the code nor to try some tweaks.
The fix you proposed would be enough, although giving that other people are facing the same issue there's hope that the official app will hopefully be somehow fixed soon.

**Comment by jakobheyman** [comment]
Here is a discussion about the 20 min averaging for the patched Libre: #2653

**Reply by enpa23** [reply]
Thank you for the hint, looks like the averaging function points here:
https://github.com/NightscoutFoundation/xDrip/blob/b58ca21907247af952a8a933b0e93c2eb55ef3bf/app/src/main/java/com/eveningoutpost/dexdrip/Models/Libre2RawValue.java#L33
I will give a try rebuilding xDrip changing 20 to 10 or so.

**Reply by jakobheyman** [reply]
Yes - I guess changing 20 to a lower number should reduce the delay.
